### ISSUE TITLE
Block() won't delete friendships and requests

### ISSUE THREAD

**Issue by leprekon91** [copleykj/socialize-user-blocking#9]
In a request Item component I have this onClick handler:
```
onClick={ () => {
    r.requester().block();
}}
```
The click inserts the block, but does not delete the request, or friendship if exists.
From the database:
```
// I have the Block:
meteor:PRIMARY> db['socialize:blocks'].find().pretty()
{
	"_id" : "MemYR88jGXyWMCLdw",
	"blockedUserId" : "mNGGQzaHuqbzP3Mpb",
	"userId" : "4KSdEt3nD6C6uTRKM",
	"createdAt" : ISODate("2019-10-03T17:05:46.453Z")
}

// Yet the Request still exists:
meteor:PRIMARY> db['socialize:requests'].find().pretty()
{
	"_id" : "KvB3iyD6H3WbzwCMz",
	"linkedObjectId" : "4KSdEt3nD6C6uTRKM",
	"objectType" : "users",
	"type" : "friend",
	"requesterId" : "mNGGQzaHuqbzP3Mpb",
	"createdAt" : ISODate("2019-10-03T17:05:23.405Z"),
	"deniedAt" : ISODate("2019-10-03T17:05:46.454Z")
}
```
Is there something I am missing?

**Comment by leprekon91** [comment]
It seems that the User.onBlock() function won't run when the block is performed. Do I need to enable it somehow? Or maybe publish some additional data?

**Comment by copleykj** [comment]
This is very strange. The only thing I can think of is that possibly there is an issue with the collection-hooks package.  I will look into it.

**Comment by copleykj** [comment]
Was this ever figured out? I know it's been awhile but can you add a list of your package versions, and is a reproduction possible?

@StorytellerCZ do you know of any issues with updates to the collection hooks package that might have caused this?

**Comment by StorytellerCZ** [comment]
Run into something similar in the voteable package recently. I think the issue is that if this is triggered on client, the server will reject any attempts at deleting anything, so you need to call a method and call the method on the server. Alternatively maybe adjusting allow/deny rules could fix this?

**Comment by copleykj** [comment]
> I think the issue is that if this is triggered on client, the server will reject any attempts at deleting anything, so you need to call a method and call the method on the server.

This should not be the case. The flow here should be:

1. `UserInstance.block()` is called which inserts a new block document on the client.
2. This change is propagated to the server where having passed allow/deny and simple-schema checks, the document is inserted into mongo.
3. The successful insert on the server causes an `insert.after` hook to run that executes all registered hooks.
4 The hook registered in the friendships package executes, deleting any current friend records or open requests between users.

**Comment by copleykj** [comment]
I believe this should be fixed  by copleykj/socialize-friendships@b1cb3cb46c279db9e4d4b65bfc108cdb250f89ad
